nms/app/components/CustomMetrics.tsx

```diff
@@ -10,37 +10,39 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import 'chartjs-adapter-date-fns';
 import React from 'react';
-import moment from 'moment';
-
+import intervalToDuration from 'date-fns/intervalToDuration';
 import {Bar, Line} from 'react-chartjs-2';
 import {ChartData, ChartTooltipItem, TimeUnit} from 'chart.js';
+import {add, isBefore, toDate} from 'date-fns';
 
 export function getStepString(delta: number, unit: string) {
   return delta.toString() + unit[0];
 }
 
-export function getStep(
-  start: moment.Moment,
-  end: moment.Moment,
-): [number, TimeUnit, string] {
-  const d = moment.duration(end.diff(start));
-  if (d.asMinutes() <= 60.5) {
+export function getStep(start: Date, end: Date): [number, TimeUnit, string] {
+  const duration = intervalToDuration({start, end});
+  if (duration.minutes && duration.minutes <= 60.5) {
```

Comment:
Are these checks like `duration.minutes` and `duration.hours` supposed to handle the null case to make typescript happy or are they also supposed to be false when e.g. `minutes` is zero? I'm not sure if all cases are handled correctly here, for example duration = 10 seconds?

Reply:
The typescript compiler is indeed happy here (there would be CI failures and comments otherwise). To make the compiler happy, I needed to extend the `if` to first check for truthy of `duration.minutes` before comparing it to a number.
The comparison intervals were not changed, as they have been as is before.

Reply:
What I mean is if we changed the logic here to make the typescript compiler happy. 0 minutes are falsy but would have been <= 60.5 in the old logic.

Reply:
Now I see what you mean. Fair. :eyes:

Reply:
Easiest fix (and somewhat cleanup) is a inversion of the logic here.